experimentation/src/main/java/com/automattic/android/experimentation/ExPlat.kt

```diff
@@ -74,20 +81,28 @@ class ExPlat(
     }
 
     private fun getAssignments(refreshStrategy: RefreshStrategy): Assignments {
-        val cachedAssignments = experimentStore.getCachedAssignments() ?: Assignments()
-        if (refreshStrategy == ALWAYS || (refreshStrategy == IF_STALE && cachedAssignments.isStale())) {
+        val cachedAssignments: Assignments =
+            runBlocking { repository.getCached() } ?: Assignments(emptyMap(), 0, 0)
```

Comment:
Re: `runBlocking`

We run file-based operation on the caller thread (risking it's a main thread). It was happening before with `experimentStore#getCachedAssignments`, but now it's just more visible.

I plan to address this in a future PR.

Reply:
> I plan to address this in a future PR.

Thanks for taking a note on that and trying to make this more "thread" robust! 🙏